app/forms/register_user_email_form.rb

```diff
@@ -20,30 +20,44 @@ def submit(params)
     user.email = params[:email].downcase
 
     if valid_form?
+      @success = true
       user.save!
     else
       process_errors
     end
+
+    result
   end
 
   private
 
+  attr_reader :success
+
   def valid_form?
     valid? && !email_taken?
   end
 
+  def result
+    {
+      success: success,
+      errors: errors.messages.values.flatten,
+      email_already_exists: email_taken?,
+      user_id: existing_user&.uuid
+    }
+  end
+
   def process_errors
     # To prevent discovery of existing emails, we check to see if the email is
     # already taken and if so, we act as if the user registration was successful.
     if email_taken? && user_unconfirmed?
       existing_user.send_confirmation_instructions
-      return true
+      return @success = true
```

Comment:
the control flow + side effect variable assignment is jumping out at me as a little strange. How would you feel about doing the assignment outside of this method and simplifying the control flow here with an `else`?

```ruby
def submit(params)
  # ...
  @success = process_errors
end

def process_errors
  if # ...
    # ...
    true
  elsif email_taken? # ...
    # ...
    true
  else
    false
  end
end
```